pkg/operator/ceph/cluster/crash/keyring.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crash
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/config/keyring"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	crashKeyringTemplate = `
+[client.crash]
+	key = %s
+	caps mon = "allow profile crash"
+	caps mgr = "allow profile crash"
+`
+)
+
+// CreateCrashCollectorSecret creates the Kubernetes Crash Collector Secret
+func CreateCrashCollectorSecret(context *clusterd.Context, clusterName string, ownerRef *metav1.OwnerReference) error {
+	k := keyring.GetSecretStore(context, clusterName, ownerRef)
+
+	// Create CrashCollector Ceph key
+	crashCollectorSecretKey, err := createCrashCollectorKeyring(k)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %q ceph keyring", crashCollectorKeyringUsername)
+	}
+
+	// Create or update Kubernetes CSI secret
+	if err := createOrUpdateCrashCollectorSecret(clusterName, crashCollectorSecretKey, k, ownerRef); err != nil {
+		return errors.Wrapf(err, "failed to create kubernetes csi secret")
+	}
+
+	return nil
+}
+
+func cephCrashCollectorKeyringCaps() []string {
+	return []string{
+		"mon", "allow profile crash",
+		"mgr", "allow profile crash",
+	}
+}
+
+func createCrashCollectorKeyring(s *keyring.SecretStore) (string, error) {
+	key, err := s.GenerateKey(crashCollectorKeyringUsername, cephCrashCollectorKeyringCaps())
+	if err != nil {
+		return "", err
+	}
+
+	return key, nil
+}
+
+func createOrUpdateCrashCollectorSecret(namespace, crashCollectorSecretKey string, k *keyring.SecretStore, ownerRef *metav1.OwnerReference) error {
```

Comment:
Don't we have a helper for creating secrets in k8sutil? Seems like we have this pattern in a bunch of places.

Reply:
I don't think we do.